bitnami/appsmith/templates/rts/deployment.yaml

```diff
@@ -0,0 +1,173 @@
+  {{- end }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: rts
+  template:
+    metadata:
+      {{- if .Values.rts.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.rts.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
+      labels: {{- include "common.labels.standard" . | nindent 8 }}
+        app.kubernetes.io/component: rts
+        {{- if .Values.rts.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.rts.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "appsmith.serviceAccountName" . }}
+      {{- include "appsmith.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.rts.hostAliases }}
+      hostAliases: {{- include "common.tplvalues.render" (dict "value" .Values.rts.hostAliases "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.rts.affinity }}
+      affinity: {{- include "common.tplvalues.render" ( dict "value" .Values.rts.affinity "context" $) | nindent 8 }}
+      {{- else }}
+      affinity:
+        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.rts.podAffinityPreset "component" "rts" "context" $) | nindent 10 }}
+        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.rts.podAntiAffinityPreset "component" "rts" "context" $) | nindent 10 }}
+        nodeAffinity: {{- include "common.affinities.nodes" (dict "type" .Values.rts.nodeAffinityPreset.type "key" .Values.rts.nodeAffinityPreset.key "values" .Values.rts.nodeAffinityPreset.values) | nindent 10 }}
+      {{- end }}
+      {{- if .Values.rts.nodeSelector }}
+      nodeSelector: {{- include "common.tplvalues.render" ( dict "value" .Values.rts.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.rts.tolerations }}
+      tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.rts.tolerations "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.rts.priorityClassName }}
+      priorityClassName: {{ .Values.rts.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.rts.schedulerName }}
+      schedulerName: {{ .Values.rts.schedulerName | quote }}
+      {{- end }}
+      {{- if .Values.rts.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.rts.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.rts.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.rts.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.rts.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.rts.terminationGracePeriodSeconds }}
+      {{- end }}
+      initContainers:
+        {{- if not .Values.diagnosticMode.enabled }}
```

Comment:
Why is it not checked if the value of volumePermissions is enabled? In the backend deployment it is checked.

Reply:
Because the RTS deployment does not have persistence, therefore no PVCs and so the volumePermissions init container does not add value.